### PR TITLE
refactor: make schema in ContentSerdes optional

### PR DESCRIPTION
**Pull request by danielpeintner**
Note: TD might not contain schema information

fixes https://github.com/eclipse/thingweb.node-wot/issues/753

### CODE REVIEW

**Comment by relu91** [comment]
I would put on hold and wait for #758

**Comment by relu91** [comment]
Fixed merge conflicts. I think we can go ahead 👍🏻 .

File: packages/core/src/codecs/octetstream-codec.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export default class OctetstreamCodec implements ContentCodec {
         }
 
         let dataLength = bytes.length;
-        let dataType: string = schema.type;
+        let dataType: string = schema ? schema.type : "undefined";
```
Comment (JKRhb) [inline packages/core/src/codecs/octetstream-codec.ts:69]: I think assignments like these could be replaced like so:

```suggestion
        let dataType: string = schema ?? "undefined";
```

**Review by danielpeintner — COMMENTED** [review]
I am fine with and without the proposed change by @JKRhb

**Comment by JKRhb** [comment]
> @relu91 @JKRhb do you have strong preferences? I am inclined to just move on with this PR...

Sure, feel free to merge from my part :)

**Pull request merged**